.gitmodules

```diff
@@ -16,3 +16,6 @@
 [submodule "plugins/obs-websocket"]
 	path = plugins/obs-websocket
 	url = https://github.com/obsproject/obs-websocket.git
+[submodule "plugins/carla/carla"]
+	path = plugins/carla/carla
+	url = https://github.com/falkTX/Carla.git
```

Comment:
https://github.com/obsproject/obs-studio/discussions/5074#discussioncomment-5837966

Just a reminder that the final version must not rely on Carla as a submodule:
- For Windows and macOS, Carla library must be provided through [obs-deps](https://github.com/obsproject/obs-deps)
- For Ubuntu, through Ubuntu official package, if not possible until newer Ubuntu version. The plugin will be disabled for Ubuntu. (The same can be said to other Linux distro known as unofficial builds).
- For the Flatpak, Carla library must be provide by adding a new module to the Flatpak manifest.

Reply:
this is a problem, we are not using carla as library but using its source code directly. it is not possible to have carla as library in the setup I have at the moment, because the intention is to integrate it into OBS as much as possible.
plus Carla UI is written in python, with Qt5 being the target at the moment and not ready for Qt6 alike OBS.

I have made it so that code on carla side is tweaked specially for OBS (for example, hiding plugins with CV ports or that have more than 8 audio IO, which wont work properly).
The build is also specialized for OBS, taking out any feature that is unnecessary for the usage of plugins within OBS.

How would this be achieved via doing the same via obs-deps? it would basically just be a "source repo" at the end...

Reply:
> > * List of VST3 folders is not configurable for now, it will just use the common folders from VST3 spec
> 
> From a quick test, the Flatpak does not show VST_PATH plugins (I have a LinuxAudio extension that provide VSTs (2 and 3)), the plugin must rely on such env vars.

the env var to use here is `VST3_PATH`, as `VST_PATH` typically referes to VST2 plugins (we should not mix the 2).
not that steinberg cares though, https://github.com/steinbergmedia/vst3sdk/issues/67 is still open after all.

but try with `VST3_PATH` and let me know if that works for you

Reply:
> the env var to use here is `VST3_PATH`, as `VST_PATH` typically referes to VST2 plugins (we should not mix the 2). not that steinberg cares though, [steinbergmedia/vst3sdk#67](https://github.com/steinbergmedia/vst3sdk/issues/67) is still open after all.
> 
> but try with `VST3_PATH` and let me know if that works for you

The manifest then needs to be updated to support VST 3, it requires me a pull the branch and modify the manifest to test then.

Reply:
> this is a problem, we are not using carla as library but using its source code directly. it is not possible to have carla as library in the setup I have at the moment, because the intention is to integrate it into OBS as much as possible. plus Carla UI is written in python, with Qt5 being the target at the moment and not ready for Qt6 alike OBS.
> 
> I have made it so that code on carla side is tweaked specially for OBS (for example, hiding plugins with CV ports or that have more than 8 audio IO, which wont work properly). The build is also specialized for OBS, taking out any feature that is unnecessary for the usage of plugins within OBS.
> 
> How would this be achieved via doing the same via obs-deps? it would basically just be a "source repo" at the end...

If it is just using bits and pieces of Carla (and not the entire project compiled as a library), it would be a better idea to just incorporate the code into the plugin itself and not rely on Carla at all.

At first glance it seems that quite a lot of plumbing went into getting just the right parts of Carla source code bundled into an assortment of static libraries, which are then linked to the plugin and creates a bit of a messy code architecture and structure (also pollutes the target/schema tree a bit).

So either Carla needs to be shaped into a library that can be used directly by third parties such as an OBS plugin (so that we can build and ship it via `obs-deps` or make it available as a Debian package), _or_ the necessary code gets copied directly into this plugin's source tree and cleaned up (i.e., all parts not required are removed, possibly restructured to remove number of compilation units).

Reply:
I almost didnt see this with all the messages going around...

Getting carla as a library is a little tricky as I am using not just the plugin hosting part, but parts of the frontend too for the plugin list dialog. Even though the code is already modular, it would be quite the restructure to get some things be a library with others being internal, plus break compatibility with existing projects that already use Carla.

So I think the best approach would be to simply copy the relevant code, keeping only the parts that are necessary.
But I would do it in some way that it is possible to sync back the changes made from OBS into Carla again. So fixes on one side benefit the other, in both ways.

The only possible problem is of code styling, where imported code does not match the style used by OBS clang-format, can an exception be made here?

Reply:
an update on this thread because someone might read it and it passes old information..
I am attempting to get stuff building with carla as dependency from `obs-deps` as requested, after seeing that if import the UI code into OBS the whole thing becomes tenable.